airbyte-protocol/models/src/main/resources/airbyte_protocol/airbyte_protocol.yaml

```diff
@@ -52,15 +52,27 @@ definitions:
       - emitted_at
     properties:
       stream:
-        description: "the name of the stream for this record"
+        description: Name of the stream for this record
         type: string
       data:
-        description: "the record data"
+        description: The data
         type: object
         existingJavaType: com.fasterxml.jackson.databind.JsonNode
       emitted_at:
-        description: "when the data was emitted from the source. epoch in millisecond."
+        description: When the data was emitted from the source. (epoch in milliseconds)
         type: integer
+      metadata:
+        "$ref": "#/definitions/AirbyteRecordMessageMetadata"
+  AirbyteRecordMessageMetadata:
+    description: "Map of metadata fields. Keys and values must both be strings. Below are some well-known fields.\n
+     replication_batch_id: Identifies which sync job replicated the record\n
+     cdc_deleted_at: Timestamp when the record was deleted. null or unset if it has not been deleted. Only present for streams syncing data using CDC. (epoch in milliseconds)\n
+     cdc_updated_at: Timestamp when the record was updated. Only present for streams syncing data using CDC. (epoch in milliseconds)\n
+     cdc_lsn: Identifies the transaction that produced the contained record. Only present for streams syncing data using CDC.\n
+     "
+    type: object
+    additionaProperties:
```

Comment:
why not encode the properties in the schema?

Reply:
Yeah. So I think this is the interesting question in this PR. I think we can do 1 of 2 things:
1. we encode the properties in the schema (and do not allow any non named metadata fields).
2. we do not encode the properties and treat all metadata fields the same.

I think the middle ground is pretty confusing (though technically also an option).

Reply:
The benefit of 2 is that if you squint destinations are still CDC-unaware (Singer destinations are also CDC-unaware). They just have a simple contract on how they support all metadata fields. Option 1 is certainly more concrete in the protocol, but now makes destinations CDC aware.

Reply:
what's the delta in CDC-awareness if `additionalProperties` is true and all metadata fields are optional? It seems like it's just a matter of how formally it's documented in the types. If I'm not missing something here I prefer 1